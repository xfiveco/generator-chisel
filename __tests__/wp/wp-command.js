const { answers } = require('./helpers');

describe('WP wp command', () => {
  test('Works OK and passes error code on errors', async () => {
    await global.chiselTestHelpers.generateProjectWithAnswers(
      ['create'],
      answers(),
      { interceptWpConfig: true },
    );

    jest.doMock('execa', () =>
      jest.fn((...args) => {
        if (typeof args[2] === 'object' && args[2].stdio === 'inherit') {
          args[2].stdio = 'pipe';
        }
        const execa = jest.requireActual('execa');
        return execa(...args);
      }),
    );

    jest.spyOn(process, 'exit').mockImplementation((...args) => {
      if (args[0] && args[0] !== 0) {
        throw new Error(`Process.exit: ${args.join(', ')}`);
      }
    });

    const mockArgv = jest
      .spyOn(process.argv, 'slice')
      .mockImplementationOnce(() => ['theme', 'list', '--format=csv']);

    await global.chiselTestHelpers.runChiselScript(['wp']);

    const mockedExeca = require('execa');

    expect(
      await mockedExeca.mock.results[0].value.then((v) => ({
        stdout: v.stdout,
        stderr: v.stderr,
      })),
    ).toMatchSnapshot();

    mockedExeca.mockClear();

    // test not-existing command

    mockArgv.mockImplementationOnce(() => ['not-existing']);

    const run = global.chiselTestHelpers.runChiselScript(['wp']);
    await expect(run).rejects.toThrowErrorMatchingSnapshot();

    expect(
      await mockedExeca.mock.results[0].value.catch((e) => ({
        stdout: e.stdout,
        stderr: e.stderr,
      })),
    ).toMatchSnapshot();
  });
});
