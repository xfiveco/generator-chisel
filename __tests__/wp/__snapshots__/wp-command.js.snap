// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WP wp command Works OK and passes error code on errors 1`] = `
Object {
  "stderr": "",
  "stdout": "name,status,update,version
chisel-test-wp-chisel,active,none,",
}
`;

exports[`WP wp command Works OK and passes error code on errors 2`] = `"Process.exit: 1"`;

exports[`WP wp command Works OK and passes error code on errors 3`] = `
Object {
  "stderr": "[31;1mError:[0m 'not-existing' is not a registered wp command. See 'wp help' for available commands.",
  "stdout": "",
}
`;
